Poco::format compile error c++23

If you try to compile Foundation with c++23 you will get an error here
```
std::string format(const std::string& fmt, const Any& value)
{
	std::string result;
	format(result, fmt, value);
	return result;
}

void format(std::string& result, const char *fmt, const std::vector<Any>& values)
{
	format(result, std::string(fmt), values);  <----error here for more than one instance of overloaded match arguments
}
```

If it's helpful here are the candidates Clang notes for this template, with the stdlib one being the most likely interfering definition:

```In file included from /usr/include/Poco/Net/SecureSMTPClientSession.h:22:
In file included from /usr/include/Poco/Net/SMTPClientSession.h:22:
In file included from /usr/include/Poco/Net/DialogSocket.h:22:
In file included from /usr/include/Poco/Net/StreamSocket.h:23:
In file included from /usr/include/Poco/FIFOBuffer.h:26:
/usr/include/Poco/Format.h:172:2: error: call to 'format' is ambiguous
  172 |         format(result, fmt, values);
      |         ^~~~~~
/usr/include/Poco/Format.h:108:21: note: candidate function
  108 | void Foundation_API format(std::string& result, const char *fmt, const std::vector<Any>& values);
      |                     ^
/usr/include/Poco/Format.h:134:6: note: candidate function [with T = std::vector<Poco::Any>, Args = <>]
  134 | void format(std::string& result, const char* fmt, T arg1, Args... args)
      |      ^
/usr/include/Poco/Format.h:148:13: note: candidate function [with T = const char *, Args = <std::vector<Poco::Any>>]
  148 | std::string format(const std::string& fmt, T arg1, Args... args)
      |             ^
/usr/include/Poco/Format.h:112:21: note: candidate function
  112 | void Foundation_API format(std::string& result, const std::string& fmt, const std::vector<Any>& values);
      |                     ^
/usr/include/Poco/Format.h:120:6: note: candidate function [with T = std::vector<Poco::Any>, Args = <>]
  120 | void format(std::string& result, const std::string& fmt, T arg1, Args... args)
      |      ^
/usr/lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/format:4293:5: note: candidate function [with _Args = <const char *&, std::vector<Poco::Any> &>]
 4293 |     format(format_string<_Args...> __fmt, _Args&&... __args)
      |     ^```

> > If it's helpful here are the candidates Clang notes for this template, with the stdlib one being the most likely interfering definition:
> > ```
> > In file included from /usr/include/Poco/Net/SMTPClientSession.h:22:
> > In file included from /usr/include/Poco/Net/DialogSocket.h:22:
> > In file included from /usr/include/Poco/Net/StreamSocket.h:23:
> > In file included from /usr/include/Poco/FIFOBuffer.h:26:
> > /usr/include/Poco/Format.h:172:2: error: call to 'format' is ambiguous
> >   172 |         format(result, fmt, values);
> >       |         ^~~~~~
> > /usr/include/Poco/Format.h:108:21: note: candidate function
> >   108 | void Foundation_API format(std::string& result, const char *fmt, const std::vector<Any>& values);
> >       |                     ^
> > /usr/include/Poco/Format.h:134:6: note: candidate function [with T = std::vector<Poco::Any>, Args = <>]
> >   134 | void format(std::string& result, const char* fmt, T arg1, Args... args)
> >       |      ^
> > /usr/include/Poco/Format.h:148:13: note: candidate function [with T = const char *, Args = <std::vector<Poco::Any>>]
> >   148 | std::string format(const std::string& fmt, T arg1, Args... args)
> >       |             ^
> > /usr/include/Poco/Format.h:112:21: note: candidate function
> >   112 | void Foundation_API format(std::string& result, const std::string& fmt, const std::vector<Any>& values);
> >       |                     ^
> > /usr/include/Poco/Format.h:120:6: note: candidate function [with T = std::vector<Poco::Any>, Args = <>]
> >   120 | void format(std::string& result, const std::string& fmt, T arg1, Args... args)
> >       |      ^
> > /usr/lib/gcc/x86_64-linux-gnu/14/../../../../include/c++/14/format:4293:5: note: candidate function [with _Args = <const char *&, std::vector<Poco::Any> &>]
> >  4293 |     format(format_string<_Args...> __fmt, _Args&&... __args)
> >       |     ^```
> > ```
> 
> Hi [@seektechnz](https://github.com/seektechnz), did you compile code from main branch? If so, provide complete compile output, please.

@matejk, I've modified the CMakeLists for building just Poco and the error doesn't exist. It only exists when importing headers directly into a project.

--- a/Foundation/include/Poco/Format.h
+++ b/Foundation/include/Poco/Format.h
@@ -118,14 +118,14 @@ inline void formatAny(std::string& result, const std::string& fmt, const std::ve
 	/// Supports a variable number of arguments and is used by
 	/// all other variants of format().
 {
-	Poco::format(result, fmt, values);
+	::Poco::format(result, fmt, values);
 }
 
 inline void formatAny(std::string& result, const char *fmt, const std::vector<Any>& values)
 	/// Supports a variable number of arguments and is used by
 	/// all other variants of format().
 {
-	Poco::format(result, fmt, values);
+	::Poco::format(result, fmt, values);
 }
 
 template <typename T, typename... Args>

